HIPs/firefly.md

```diff
@@ -0,0 +1,374 @@
+
+#### Integration of core blockchain constructs
+
+Core blockchain constructs are composable building blocks of multi-party solutions, and can coexist seamlessly with traditional private messaging and back-office system integration.
+
+- Tokens
+  - Non-fungible tokens (NFTs) - universally unique
+  - Fungible tokenized value exchange - total conversation of value
+- Signatures
+  - Attestation to data, at a particular point in time
+- Multi-party event sequencing
+  - The game-changing feature of blockchain for enterprise multi-party systems
+- Deterministic logic
+  - Custom on-chain smart contracts
+  - Trusted execution environments (TEEs)_ - plug point_
+  - Zero-knowledge Proofs (ZKPs) - _plug point_
+  - Multi-party Compute (MPC) - _plug point_
+
+#### Coordination of on-chain and off-chain interactions
+
+Sometimes underestimated is the importance, and complexity, of synchronizing the exchange of on-chain and off-chain data. Yet tackling this problem is almost universal in enterprise multi-party system development.
+
+FireFly provides a rugged, enterprise-grade runtime and simple event-driven REST API model, so that developers can focus on solving interesting business problems - rather than re-building plumbing. Second order problems are also addressed, like high throughput through batching of on-chain transactions, and providing n-way application scale while retaining deterministic ordering.
+
+The pluggable architecture allows the underlying infrastructure components to be swapped out, without changing the application. That includes the database, blob store, private messaging technology, and even the blockchain itself.
+
+![Event Sequencing](../assets/event-sequencing.svg "Event Sequencing")
+
+![Ping Pong](../assets/ping-pong.svg "Ping Pong")
+
+### Pre-built implementations for the big three enterprise blockchain platforms
+
+Remote agents for the blockchain interface are provided as follows:
+
+- Enterprise ethereum (Hyperledger Besu / Quorum)
+  - REST Gateway and Event Streams via sister project: _[ethconnect](https://github.com/kaleido-io/ethconnect)_
+  - On-chain smart contracts supplied with FireFly Core
+  - Full sequencing, and token support (ERC-20 / ERC-721)
+- Hyperledger Fabric
+  - _Under development (lead by Jim Zhang)_
+- Corda
+  - REST Gateway and Event Streams via sister project: _source opening soon_
+  - On-chain smart contracts supplied with FireFly Core
+  - Implements block based sequencing of events, within a privacy group
+
+## Effort and resources
+
+The FireFly code base’s current version is in production with multiple large consortiums, so Kaleido is fully committed to continued support for the open source project with currently 6 developers working full time on the code base.
```

Comment:
Can you split it out by implementation here. When was the typescript implementation written and released (I believe this is known as KAT)? When was the Go implementation written and released (I believe this is known as FireFly)? How many releases have there been of each? Are there production users on both? Are they both intended to move forward as part of the FireFly proposal?